ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_averages_and_variances.py

```diff
@@ -354,3 +354,47 @@ def test_numpy_nanpercentile(
         test_flags=test_flags,
         input_dtypes=input_dtypes,
     )
+
+
+# nanmedian
+@handle_frontend_test(
+    fn_tree="numpy.nanmedian",
+    dtype_and_a=statistical_dtype_values(function="mean"),
+    dtype=helpers.get_dtypes("float", full=False, none=True),
+    where=np_frontend_helpers.where(),
+    keep_dims=st.booleans(),
+)
+def test_numpy_nanmedian(
+    dtype_and_a,
+    dtype,
+    where,
+    frontend,
+    test_flags,
+    fn_tree,
+    on_device,
+    keep_dims,
+):
+    input_dtypes, a, axis = dtype_and_a
+    if isinstance(axis, tuple):
+        axis = axis[0]
+
+    where, input_dtypes, test_flags = np_frontend_helpers.handle_where_and_array_bools(
+        where=where,
+        input_dtype=input_dtypes,
+        test_flags=test_flags,
+    )
+
+    np_frontend_helpers.test_frontend_function(
+        input_dtypes=input_dtypes,
+        frontend=frontend,
+        test_flags=test_flags,
+        fn_tree=fn_tree,
+        on_device=on_device,
+        a=a[0],
+        axis=axis,
+        dtype=dtype[0],
+        out=None,
+        keepdims=keep_dims,
+        where=where,
+        test_values=False,
```

Comment:
All the changes are made

Reply:
> > Thanks for making those changes ! It's going in the right direction, just left some more comments. Thanks !
> 
> Where?

I have left some comments inside my code (If that what you mean)
And thanks for your patience.

Reply:
Are there any guides or tutorials to help me add testing strategies for overwrite_input and axis?

Reply:
> You are correct that the tests fail because of nanpercentile and you don't need to fix that for this PR as it is unrelated to your function. I think this looks mostly good, a small detail left which causes the lint tests to fail is that you've passed `overwrite_input=overwrite_input` in the `test_numpy_nanmedian` function definition when it should just be `overwrite_input`, if you can fix this then the PR should be good to merge I think

Done

Reply:
> The tests fail due to the axis input being incorrectly handled, on second reading I realized I misread one of the test function inputs you used. `dtype_x_axis` already takes into account the axis variable so there is no need to explicitly implement a separate search strategy for `axis`. Could you try to revert back and remove lines 364 (`axis=st.integers()`), 373 (`axis,`) and replace `axis2` by `axis` in line 376 ? Hopefully this should fix the issue. Apologies for the mistake !

No, It's okay
I have learned a lot from this journey ^^